Deprecate TI cc2650

vanti opened the pull request:
Adding Kconfig settings to inform anyone trying to build for this
platform of its pending deprecation in 2.2.0.

vanti committed:
soc: kconfig: add SOC_DEPRECATED_RELEASE

Adding a Kconfig parameter so that we can indicate an SoC is to be
deprecated, similar to what is being done for BOARD_DEPRECATED_RELEASE.

Signed-off-by: Vincent Wan <vincent.wan@linaro.org>

vanti committed:
kconfig: deprecate TI cc2650_sensortag and cc2650 SoC

Adding Kconfig settings to warn anyone trying to build for this
platform of its pending deprecation in 2.2.0.

Signed-off-by: Vincent Wan <vincent.wan@linaro.org>

ioannisg commented on soc/Kconfig:
```diff
@@ -71,3 +71,11 @@ config SOC_RWDATA_LD
+config SOC_DEPRECATED
+	string
+	help
+	  This hidden option is set in the soc configuration and indicates
```
SoC

ioannisg replied:
similar entries below

vanti replied:
Thanks, fixed.

ioannisg reviewed (commented):
Shouldn't we have BOARD_DEPRECATED_RELEASE, and SOC_DEPRECATED_RELEASE, to indicated better that the string element refers to the release? Or this terminilogy has been agreed in the API working group?

vanti committed:
boards: Kconfig: rename BOARD_DEPRECATED to BOARD_DEPRECATED_RELEASE

Renaming this Kconfig option given it corresponds to a version string
for a particular release.

Signed-off-by: Vincent Wan <vincent.wan@linaro.org>

vanti commented:
> Shouldn't we have BOARD_DEPRECATED_RELEASE, and SOC_DEPRECATED_RELEASE, to indicated better that the string element refers to the release? Or this terminilogy has been agreed in the API working group?

Not sure about the history on this, but I like your names better. Given I don't see anyone else using this Kconfig option in existing code, I went ahead and renamed them.
